### PR TITLE
16 ability to register package routes in required place of the app

### PR DESCRIPTION
**Pull request by RGO230**

### CODE REVIEW

**Comment by RGO230** [comment]
commit has merge branches, because they need for testing this functional

File: src/ChatServiceProvider.php [hunk 1]
```diff
@@ -30,6 +31,10 @@ class ChatServiceProvider extends ServiceProvider
 {
     public function boot()
     {
+        Route::mixin(new ChatRouter());
+
+        Route::macro('when', fn ($condition, $callback) => ($condition) ? $callback() : null);
```
Comment (DenTray) [inline src/ChatServiceProvider.php:36]: let's move `when` route macros to the laravel helpers package

**Review by DenTray — CHANGES_REQUESTED** [review]

File: composer.json [hunk 2]
```diff
@@ -13,8 +13,9 @@
         }
     ],
     "require": {
-        "php": ">=7.4.0",
-        "ronasit/laravel-helpers": "^2.1"
+        "php": "^8.3",
+        "laravel/framework": "^11.21",
```
Comment (DenTray) [inline composer.json:17]: let's use 30 minor version 

```suggestion
        "laravel/framework": "^11.30",
```

File: config/chat.php [hunk 3]
```diff
@@ -7,7 +7,7 @@
 return [
     'classes' => [
         'user_model' => User::class,
-        'media_model' => Media::class,
+        'media_model' => RonasIT\Media\Models\Media::class,
```
Comment (DenTray) [inline config/chat.php:10]: let's remove it from this branch

**Review by DenTray — CHANGES_REQUESTED** [review]

File: config/chat.php [hunk 4]
```diff
@@ -1,13 +1,11 @@
 <?php
 
 use App\Models\User;
-use App\Models\Media;
```
Comment (DenTray) [inline config/chat.php:4]: please revert these changes

File: src/ChatServiceProvider.php [hunk 5]
```diff
@@ -30,6 +31,10 @@ class ChatServiceProvider extends ServiceProvider
...
+
```
Comment (DenTray) [inline src/ChatServiceProvider.php:37]:
```suggestion
```

**Review by DenTray — CHANGES_REQUESTED** [review]

File: src/ChatRouter.php [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+                'messages_search' => $options['messages_search'] ?? true,
+                'messages_create' => $options['messages_create'] ?? true,
+                'messages_read' => $options['messages_read'] ?? true,
+            ];
+
+            $this->controller(ConversationController::class)->group(function () use ($options) {
+                when($options['conversations_search'], fn () => $this->get('conversations', 'search')->name('conversations.search'));
+                when($options['conversations_get'], fn () => $this->get('conversations/{id}', 'get')->name('conversations.get'));
+                when($options['conversations_delete'], fn () => $this->delete('conversations/{id}', 'delete')->name('conversations.delete'));
+                when($options['conversations_get_by_user_id'], fn () => $this->post('users/{userId}/conversation', 'getByUserId')->name('conversations.create.get.user.by.id'));
```
Comment (DenTray) [inline src/ChatRouter.php:33]:
```suggestion
                when($options['conversations_get_by_user_id'], fn () => $this->post('users/{userId}/conversation', 'getByUserId')->name('conversations.get_by_user_id'));
```

File: config/chat.php [hunk 7]
```diff
@@ -1,7 +1,7 @@
 <?php
 
 use App\Models\User;
-use RonasIT\Media\Models\Media;
+use App\Models\Media;
```
Comment (DenTray) [inline config/chat.php:4]:
```suggestion
RonasIT\Media\Models\Media;
```

**Review by DenTray — CHANGES_REQUESTED** [review]

File: config/chat.php [hunk 8]
```diff
@@ -1,7 +1,7 @@
 <?php
 
 use App\Models\User;
-use RonasIT\Media\Models\Media;
+RonasIT\Media\Models\Media;
```
Comment (DenTray) [inline config/chat.php:4]:
```suggestion
use RonasIT\Media\Models\Media;
```

**Review by DenTray — APPROVED** [review]

**Review by DenTray — APPROVED** [review]

**Pull request merged**